test/docker_tests/conftest.py

```
@@ -13,8 +14,48 @@ def _announce(request: pytest.FixtureRequest, message: str) -> None:
+    try:
+        shutil.rmtree(mounts_dir)
+    except PermissionError:
+        # Fallback to docker for root-owned files
+        # We mount the parent directory to delete the directory itself
+        cmd = [
+            "docker", "run", "--rm",
+            "-v", f"{project_root}:/work",
+            "alpine:3.22",
+            "rm", "-rf", "/work/test_mounts"
+        ]
+        subprocess.run(
```

Ruff (0.14.10)

`subprocess` call: check for execution of untrusted input

(S603)